rust/gui-client/src-tauri/src/client/tunnel_wrapper/windows.rs

```diff
@@ -0,0 +1,109 @@
+use anyhow::{anyhow, Context as _, Result};
+use connlib_client_shared::Callbacks;
+use firezone_headless_client::{IpcClientMsg, IpcServerMsg};
+use futures::{SinkExt, Stream};
+use secrecy::{ExposeSecret, SecretString};
+use std::{pin::pin, task::Poll};
+use tokio::{net::windows::named_pipe, sync::mpsc};
+use tokio_util::codec::{Framed, LengthDelimitedCodec};
+
+pub(crate) struct TunnelWrapper {
+    task: tokio::task::JoinHandle<Result<()>>,
+    // Needed temporarily to avoid a big refactor. We can remove this in the future.
+    tx: mpsc::Sender<String>,
+}
+
+impl TunnelWrapper {
+    pub(crate) async fn disconnect(mut self) -> Result<()> {
+        self.send_msg(&IpcClientMsg::Disconnect)
+            .await
+            .context("Couldn't send Disconnect")?;
+        self.task.abort();
+        Ok(())
+    }
+
+    #[allow(clippy::unused_async)]
+    pub(crate) async fn send_msg(&mut self, msg: &IpcClientMsg) -> Result<()> {
+        self.tx
+            .send(serde_json::to_string(msg).context("Couldn't encode IPC message as JSON")?)
+            .await
+            .context("Couldn't send IPC message")?;
+        Ok(())
+    }
+}
+
+enum IpcEvent {
+    /// The client wants to send a message to the service
+    Client(String),
+    /// The connlib instance in the server wants to send a message to the client
+    Connlib(IpcServerMsg),
+}
+
+pub(crate) async fn connect(
```

Comment:
Since this is used to create `TunnelWrapper` I think it will be clearer if the implementaiton is moved within `TunnelWrapper` name-space

Reply:
I was originally copying the connlib API but now it's moved